engine/src/main/java/org/kigalisim/engine/state/StreamParameterization.java

```diff
@@ -326,7 +388,8 @@ public void resetStateAtTimestep() {
    * @throws IllegalArgumentException If the stream name is not a sales substream
    */
   private void ensureSalesStreamAllowed(String name) {
-    if (!"domestic".equals(name) && !"import".equals(name) && !"export".equals(name) && !"recycle".equals(name)) {
+    if (!"domestic".equals(name) && !"import".equals(name) && !"export".equals(name)
```

Comment:
Let's make this is new style switch statement with a supporting private method called getIsSalesStreamAllowed.